pytorch_lightning/trainer/trainer.py

```diff
@@ -754,6 +749,10 @@ def num_gpus(self) -> int:
     def data_parallel(self) -> bool:
         return self.use_dp or self.use_ddp or self.use_ddp2
 
+    @property
+    def progress_bar_callback(self):
+        return self._progress_bar_callback
```

Comment:
not sure how much it protects as it handles pointer to the same object so edit in the return will appear in the original one...

Reply:
yes I think that's fine. the goal was to make the reference read-only so that you can't change the reference to anything other than a progress bar.